test/distributed/org/apache/cassandra/distributed/test/streaming/StreamFailureTest.java

```diff
@@ -0,0 +1,327 @@
+                                      .start())
+        {
+            init(cluster);
+
+            cluster.schemaChange(withKeyspace("CREATE TABLE %s.tbl (pk int PRIMARY KEY)"));
+
+            triggerStreaming(cluster, zeroCopyStreaming);
+            // make sure disk failure policy is not triggered
+
+            IInvokableInstance failingNode = cluster.get(failedNode);
+
+            searchForLog(failingNode, reason);
+        }
+    }
+
+    private void streamTest2(String reason) throws IOException
+    {
+        try (Cluster cluster = Cluster.build(2)
+                                      .withTokenSupplier(TokenSupplier.evenlyDistributedTokens(3))
+                                      .withInstanceInitializer(BB::install)
+                                      .withConfig(c -> c.with(Feature.values())
+                                                        // when die, this will try to halt JVM, which is easier to validate in the test
+                                                        // other levels require checking state of the subsystems
+                                                        .set("timeout_delay", "1ms"))
+                                      .start())
+        {
+
+            init(cluster);
+            cluster.schemaChange(withKeyspace("CREATE TABLE %s.tbl (pk int PRIMARY KEY)"));
+
+            ForkJoinPool.commonPool().execute(() -> triggerStreaming(cluster, true));
+            State.STREAM_IS_RUNNING.await();
+            logger.info("Streaming is running... time to wake it up");
+            State.UNBLOCK_STREAM.signal();
+
+            IInvokableInstance failingNode = cluster.get(1);
+
+            searchForLog(failingNode, reason);
+
+
+        }
+    }
+
+
+    private void triggerStreaming(Cluster cluster, boolean expectedEntireSSTable)
+    {
+        IInvokableInstance node1 = cluster.get(1);
+        IInvokableInstance node2 = cluster.get(2);
+
+        // repair will do streaming IFF there is a mismatch; so cause one
+        for (int i = 0; i < 10; i++)
+            node1.executeInternal(withKeyspace("INSERT INTO %s.tbl (pk) VALUES (?)"), i); // timestamp won't match, causing a mismatch
+
+        // trigger streaming; expected to fail as streaming socket closed in the middle (currently this is an unrecoverable event)
+        //Blocks until the stream is complete
+        node2.nodetoolResult("repair", "-full", KEYSPACE, "tbl").asserts().failure();
+    }
+
+    private void searchForLog(IInvokableInstance failingNode, String reason)
+    {
+        LogResult<List<String>> result = failingNode.logs().grepForErrors(-1, Pattern.compile("Stream failed:"));
+        // grepForErrors will include all ERROR logs even if they don't match the pattern; for this reason need to filter after the fact
+        List<String> matches = result.getResult();
+
+        matches = matches.stream().filter(s -> s.startsWith("WARN") && s.contains("Stream failed")).collect(Collectors.toList());
```

Comment:
How do you know this is your `Stream failed` message rather than something else just containing the same string?

Reply:
wouldn't `Stream failed` match everything?  We explicitly searched for that... so everything *must* contain that string.